src/Refactoring2-Transformations/RBTemporaryToInstanceVariableTransformation.class.st

```diff
@@ -100,6 +100,22 @@ RBTemporaryToInstanceVariableTransformation class >> model: aRBSmalltalk class:
 		yourself
 ]
 
+{ #category : #preconditions }
+RBTemporaryToInstanceVariableTransformation >> applicabilityConditions [ 
+
+	^(RBCondition definesSelector: selector in: class)
+		& (RBCondition definesInstanceVariable: temporaryVariableName asString in: class) not
+		& (RBCondition withBlock: [
+			(class allSubclasses anySatisfy:
+				[ :cls | cls definesInstanceVariable: temporaryVariableName asString ])
+			ifTrue: [ self refactoringWarning:
+				('One or more subclasses of <1p> already defines an<n>instance variable with the same name. Proceed anyway?' expandMacrosWith: class name) ].
+			true ])
```

Comment:
I think this can be qualified as a breaking change?

Reply:
Yes!

Reply:
In fact we cannot have a class with an instance variable with the same name it will break the system.

Reply:
So I do not understand the proceed anyway